src/ashley/machina_extensions/forum_search/forms.py

```diff
@@ -13,6 +17,23 @@ class SearchForm(MachinaSearchForm):
     even with an empty search in the main field.
     """
 
+    def __init__(self, *args, **kwargs):
+        # load current lti_context to filter forum search
+        lti_contexts = kwargs.pop("lti_contexts", None)
+        super().__init__(*args, **kwargs)
+        user = kwargs.pop("user", None)
+        self.allowed_forums = PermissionHandler().get_readable_forums(
+            Forum.objects.filter(lti_contexts=lti_contexts), user
+        )
+        if self.allowed_forums:
```

Comment:
So if I understand well, you limit the options proposed to the user in the forum filter select box so that s.he can only see allowed forums. But you are not forcing a filter to only return results within these allow forum are you?

Reply:
the search method is in MachinaSearchForm class and uses the allowed_forums so it filters on it as well, so in both case it's filtered

```
def search(self):
        sqs = super().search()

        if not self.is_valid():
            return self.no_query_found()

        # Handles topic-based searches
        if self.cleaned_data['search_topics']:
            sqs = sqs.filter(topic_subject=AutoQuery(self.cleaned_data['q']))

        # Handles searches by poster name
        if self.cleaned_data['search_poster_name']:
            sqs = sqs.filter(poster_name__icontains=self.cleaned_data['search_poster_name'])

        # Handles searches in specific forums if necessary
        if 'search_forums' in self.cleaned_data and self.cleaned_data['search_forums']:
            sqs = sqs.filter(forum__in=self.cleaned_data['search_forums'])
        else:
            forum_ids = self.allowed_forums.values_list('id', flat=True)
            sqs = sqs.filter(forum__in=forum_ids) if forum_ids else sqs.none()

        return sqs
```

Reply:
My understanding is that it filters on a forum if it is passed as parameter. Otherwise you have just limited the filter choices allowed forum but not forced it to the current forum.

Maybe what you should do is override this search method and hardcode a filter on the forums of the current context?

But first, we need to design a test that proves if my concerns are justified : a user allowed on 2 forums and accessing Ashley with a context targetting the first forum, should not see a result posted on the second forum?

Reply:
this test is one already unless i misunderstood you

Reply:
Ah ok got it.
So I think you should also test what happens if someone specifies an unauthorized forum in the `search_forums` parameter. It should return a 400 if I understand well?

Reply:
And me too I get what you are saying, is that the original code is trusting the search_forums field without checking again if it's part of the allowed forums. I will had a constraint to make this completely clean, so override the search method. 
To allow or not the access on topics that are not part of the lti context and deal with 400 request, that won't be dealt in this PR, this is relative to all the project, not just relative to the search and will be done in a separate PR.

Reply:
I think it is already taken care of because Django won't accept a field value that is now in choices? I was just asking to add a test to secure it because all these overrides are fragile.

Reply:
So I did the extra test, but maybe we misunderstood each other. I don't get a 400 but a 200 with an error message, telling me to choose a good forum because the one chosen is not part of my choices.

Reply:
ok! It's interesting that you don't get a 400 :man_shrugging: But as long as you don't access the content it should be fine.